src/containers/post/post.component.jsx

```diff
@@ -4,11 +4,36 @@ import { fetchPostUser, fetchPostUserAvatar } from "database/postUser";
 import { useSelector } from "react-redux";
 
 const Post = (props) => {
+  const {
+    optionA,
+    optionB,
+    body,
+    ownerID,
+    hasUserVotedForA,
+    hasUserVotedForB,
+    id,
+    optionAName,
+    optionBName,
+  } = props;
   const [canUserViewVote, setCanUserViewVote] = useState(false);
-  const [voteACount, setVoteACount] = useState(props.optionA.length);
-  const [voteBCount, setVoteBCount] = useState(props.optionB.length);
+  const [voteACount, setVoteACount] = useState(optionA.length);
+  const [voteBCount, setVoteBCount] = useState(optionB.length);
+  const [viewTotalVotes, setViewTotalVotes] = useState(false);
+  const [postUserUsername, setPostUserUsername] = useState("");
+  const [postUserAvatar, setPostUserAvatar] = useState("");
 
   const userIDFromState = useSelector((state) => state.users.user.userId);
+  const voteAPercent = (voteACount / (voteACount + voteBCount)) * 100;
+  const voteBPercent = (voteBCount / (voteACount + voteBCount)) * 100;
+
+  useEffect(() => {
+    fetchPostUser(ownerID).then((result) => {
```

Comment:
This is a nitpick, but as opposed to `result` or `res` always a nice idea to name it what it actually is so that it is more readable. For example, in this case might be something like `userObject`

Reply:
Makes sense